components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -342,7 +352,7 @@ class SystemEngineSession(
     override fun toggleDesktopMode(enable: Boolean, reload: Boolean) {
         val webSettings = webView.settings
         webSettings.userAgentString = toggleDesktopUA(webSettings.userAgentString, enable)
-        webSettings.useWideViewPort = enable
+        webSettings.useWideViewPort = settings.useWideViewPort ?: enable
```

Comment:
My assumption here is that for `enable = true` we will always need to enable `useWideViewPort` (otherwise desktop mode is just broken). Only for `enable = false` we need to either fall back to the consumer's choice  or `false`.

Reply:
This also implies for `toggleDesktopUA` and `notifyObserver` condition right?

Reply:
Yes, the UA will continue to toggle like before :)